starter-cli/src/test/groovy/io/micronaut/starter/cli/CodeGenConfigSpec.groovy

```diff
@@ -15,7 +16,9 @@ import java.nio.file.Files
 
 class CodeGenConfigSpec extends Specification {
 
-    @Shared @AutoCleanup BeanContext ctx = BeanContext.run()
+    @Shared
+    @AutoCleanup
+    ApplicationContext ctx = ApplicationContext.run()
```

Comment:
If I leave `BeanContext` fort his test I get: 


Failed to inject value for parameter [environment] of class: io.micronaut.http.resource.ResourceLoaderFactory

Message: No bean of type [io.micronaut.context.env.Environment] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new DefaultAvailableFeatures([List features]) --> new Maven([MavenBuildCreator dependencyResolver]) --> new MavenBuildCreator([MavenCoordinateResolver mavenCoordinateResolver],PropertiesResolver propertiesResolver) --> new MavenCoordinateResolver([DependencyVersionResolver[] dependencyVersionResolvers],PropertiesResolver propertiesResolver) --> new PomDependencyVersionResolver([ResourceResolver resourceResolver]) --> new ResourceLoaderFactory([Environment environment])
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [environment] of class: io.micronaut.http.resource.ResourceLoaderFactory

Message: No bean of type [io.micronaut.context.env.Environment] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
Path Taken: new DefaultAvailableFeatures([List features]) --> new Maven([MavenBuildCreator dependencyResolver]) --> new MavenBuildCreator([MavenCoordinateResolver mavenCoordinateResolver],PropertiesResolver propertiesResolver) --> new MavenCoordinateResolver([DependencyVersionResolver[] dependencyVersionResolvers],PropertiesResolver propertiesResolver) --> new PomDependencyVersionResolver([ResourceResolver resourceResolver]) --> new ResourceLoaderFactory([Environment environment])
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1035)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2705)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2691)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2363)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1206)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2705)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2691)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2363)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3048)
	at io.micronaut.context.DefaultBeanContext.getBeansOfTypeInternal(DefaultBeanContext.java:2949)
	at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:1045)
	at io.micronaut.context.AbstractBeanDefinition.lambda$getBeansOfTypeForConstructorArgument$9(AbstractBeanDefinition.java:1143)
	at io.micronaut.context.AbstractBeanDefinition.resolveBeanWithGenericsFromConstructorArgument(AbstractBeanDefinition.java:1822)
	at io.micronaut.context.AbstractBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractBeanDefinition.java:1138)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:987)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2705)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2691)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2363)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2705)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2691)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2363)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3048)
	at io.micronaut.context.DefaultBeanContext.getBeansOfTypeInternal(DefaultBeanContext.java:2949)
	at io.micronaut.context.DefaultBeanContext.getBeansOfType(DefaultBeanContext.java:1045)
	at io.micronaut.context.AbstractBeanDefinition.lambda$getBeansOfTypeForConstructorArgument$9(AbstractBeanDefinition.java:1143)
	at io.micronaut.context.AbstractBeanDefinition.resolveBeanWithGenericsFromConstructorArgument(AbstractBeanDefinition.java:1822)
	at io.micronaut.context.AbstractBeanDefinition.getBeansOfTypeForConstructorArgument(AbstractBeanDefinition.java:1138)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:990)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1924)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2705)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2691)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2363)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:721)
	at io.micronaut.starter.cli.CodeGenConfig.load(CodeGenConfig.java:155)
	at io.micronaut.starter.cli.CodeGenConfigSpec.test config is compatible with the old format(CodeGenConfigSpec.groovy:35)
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.context.env.Environment] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2348)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	... 51 more

Reply:
Yep to be expected. The environment, etc.. is not part of the core bean context